io/io/src/TBufferMerger.cxx

```diff
@@ -81,10 +80,7 @@ void TBufferMerger::WriteOutputFile()
             fQueue.pop();
          }
 
-         if (!buffer) {
-            done = true;
-            break;
-         }
+         if (!buffer) goto end;
```

Comment:
Hm, that's very non-structural :) Why not just break here? Are you trying to avoid running destructors of the nested scope?